src/iidm/ThreeWindingsTransformer.cpp

```diff
@@ -18,141 +18,149 @@ namespace iidm {
 
 ThreeWindingsTransformer::ThreeWindingsTransformer(const std::string& id, const std::string& name, bool fictitious, Leg&& leg1, Leg&& leg2, Leg&& leg3, double ratedU0) :
     Connectable(id, name, fictitious, ConnectableType::THREE_WINDINGS_TRANSFORMER),
-    m_leg1(std::move(leg1)),
-    m_leg2(std::move(leg2)),
-    m_leg3(std::move(leg3)),
     m_ratedU0(ratedU0) {
-    m_leg1.setTransformer(*this);
-    m_leg2.setTransformer(*this);
-    m_leg3.setTransformer(*this);
+    m_legs.emplace_back(std::move(leg1));
+    m_legs.emplace_back(std::move(leg2));
+    m_legs.emplace_back(std::move(leg3));
+    for (Leg& leg : m_legs) {
+        leg.setTransformer(*this);
+    }
```

Comment:
Could you check if it's possible to do the job without the loop? Maybe with something like this:
```
m_legs.emplace_back(std::move(leg1.setTransformer(*this));
```
It should be OK if `setTransformer` returns the leg.